test/integration/acceptance/create_asset_test.cpp

```diff
@@ -0,0 +1,258 @@
+      "-startWithDash",
+      "@.is.not.allowed",
+      "no space is allowed",
+      "endWith-",
+      "label.endedWith-.is.not.allowed",
+      "aLabelMustNotExceeds63charactersALabelMustNotExceeds63characters",
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPad."
+      "maxLabelLengthIs63paddingPaddingPaddingPaddingPaddingPaddingPadP",
+      "257.257.257.257",
+      "domain#domain",
+      "asd@asd",
+      "ab..cd"};
+};
+
+/**
+ * @given some user with can_create_asset permission
+ * @when the user tries to create an asset
+ * @then asset is successfully created
+ */
+TEST_F(CreateAssetFixture, Basic) {
+  const auto asset_id = kAssetName + "#" + kDomain;
+  const auto asset_amount = "100.0";
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms({interface::permissions::Role::kCreateAsset,
+                                 interface::permissions::Role::kAddAssetQty}))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      // testing the target command
+      .sendTx(complete(baseTx().createAsset(kAssetName, kDomain, kPrecision)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      // testing that target command actually changed the state of the ledger
+      .sendTx(complete(baseTx().addAssetQuantity(asset_id, asset_amount)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * C235 Create asset with an empty name
+ * C236 Create asset with boundary values per name validation
+ * @given some user with can_create_asset permission
+ * @when the user tries to create an asset with invalid or empty name
+ * @then no asset is created
+ */
+TEST_F(CreateAssetFixture, IllegalCharactersInName) {
+  IntegrationTestFramework itf(1);
+  itf.setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); });
+  for (const auto &name : kIllegalAssetNames) {
+    itf.sendTx(complete(baseTx().createAsset(name, kDomain, kPrecision)),
+               checkStatelessInvalid);
+  }
+  itf.done();
+}
+
+/**
+ * C234 Create asset with an existing id (name)
+ * @given a user with can_create_asset permission
+ * @when the user tries to create asset that already exists
+ * @then stateful validation failed
+ */
+TEST_F(CreateAssetFixture, ExistingName) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
```

Comment:
Yes, it is. In the base case (or sunny day case), we are ensuring that everything works fine when input data is valid.

All the rest cases are checking that Iroha correctly handles invalid input. Since every other test case is checking some invalid command and needs preparations (user creation) we have to be sure that the user was created successfully and that was not the reason for target command failure. Otherwise, we can have false negatives.